test/e2e/multicast_test.go

```diff
@@ -158,7 +277,255 @@ type multicastTestcase struct {
+				},
+				Egress: []crdv1alpha1.Rule{},
+			},
+		}
+		for i, _ := range anp.ruleConfigs {
+			np.Spec.Egress = append(np.Spec.Egress, crdv1alpha1.Rule{
+				To: []crdv1alpha1.NetworkPolicyPeer{
+					{
+						IPBlock: &crdv1alpha1.IPBlock{
+							CIDR: fmt.Sprintf("%s/32", anp.ruleConfigs[i].address),
+						},
+					},
+				},
+				Ports: []crdv1alpha1.NetworkPolicyPort{
+					{
+						Protocol: &udp,
+					},
+				},
+				Action: &anp.ruleConfigs[i].action,
+				Name:   anp.ruleConfigs[i].name,
+			})
+		}
+		if _, err = k8sUtils.CreateOrUpdateANP(np); err != nil {
+			t.Fatalf("create ANP failed for ANP %s: %v", np.Name, err)
+		}
+		err = data.waitForANPRealized(t, data.testNamespace, np.Name)
+		if err != nil {
+			t.Fatalf("error when waiting for ANP %s released: %v", np.Name, err)
+		}
+	}
+
+	for _, anp := range mc.igmpANPConfigs {
+		np := &crdv1alpha1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{Namespace: data.testNamespace, Name: anp.name, Labels: map[string]string{"antrea-e2e": anp.name}},
+			Spec: crdv1alpha1.NetworkPolicySpec{
+				Priority: p10,
+				AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+					{PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"antrea-e2e": anp.appliedToPod}}},
+				},
+				Egress:  []crdv1alpha1.Rule{},
+				Ingress: []crdv1alpha1.Rule{},
+			},
+		}
+		for i := range anp.ruleConfigs {
+			rule := crdv1alpha1.Rule{
+				From: []crdv1alpha1.NetworkPolicyPeer{},
+				Protocols: []crdv1alpha1.NetworkPolicyProtocol{
+					{
+						IGMP: &crdv1alpha1.IGMPProtocol{IGMPType: &anp.ruleConfigs[i].igmpType, GroupAddress: anp.ruleConfigs[i].address},
+					},
+				},
+				Action: &anp.ruleConfigs[i].action,
+				Name:   anp.ruleConfigs[i].name,
+			}
+			if anp.ruleConfigs[i].igmpType == igmpQueryType {
+				np.Spec.Ingress = append(np.Spec.Ingress, rule)
+			} else {
+				np.Spec.Egress = append(np.Spec.Egress, rule)
+			}
+		}
+		if _, err = k8sUtils.CreateOrUpdateANP(np); err != nil {
+			t.Fatalf("create ANP failed for ANP %s: %v", np.Name, err)
+		}
+		err = data.waitForANPRealized(t, data.testNamespace, np.Name)
+		if err != nil {
+			t.Fatalf("error when waiting for ANP %s released: %v", np.Name, err)
+		}
+	}
+
+	for _, receiverConfig := range mc.receiverConfigs {
+		for _, addr := range receiverConfig.IPs {
+			go func(receiver, addr string) {
+				cmd := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d -W %d %s", 2234, mcjoinWaitTimeout, addr)}
+				data.RunCommandFromPod(data.testNamespace, receiver, mcjoinContainerName, cmd)
+			}(receiverConfig.name, addr)
+		}
+	}
+
+	time.Sleep(5 * time.Second)
+
+	var wg sync.WaitGroup
+
+	for _, senderConfig := range mc.senderConfigs {
+		for _, addr := range senderConfig.IPs {
+			wg.Add(1)
+			go func(sender, addr string) {
+				defer wg.Done()
+				cmd := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 30 -c 10 -W %d %s", 2234, mcjoinWaitTimeout, addr)}
```

Comment:
Use a variable for packets count instead of hard code?

Reply:
updated